jgrapht-core/src/main/java/org/jgrapht/alg/TransitiveReduction.java

```diff
@@ -0,0 +1,201 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* ----------------------
+ * TransitiveReduction.java
+ * ----------------------
+ *
+ * Original Author:   Christophe Thiebaud
+ * Contributor(s):
+ *
+ * Changes
+ * -------
+ * 13-August-2015: Initial revision (CT);
+ *
+ */
+package org.jgrapht.alg;
+
+import java.util.*;
+
+import org.jgrapht.*;
+
+/**
+ * An implementation of Harry Hsu's
+ * <a href="https://en.wikipedia.org/wiki/Transitive_reduction">transitive
+ * reduction algorithm</a>.
+ * 
+ * <p>
+ * cf.
+ * <a href="http://projects.csail.mit.edu/jacm/References/hsu1975:11.html">Harry
+ * Hsu. "An algorithm for finding a minimal equivalent graph of a digraph.",
+ * Journal of the ACM, 22(1):11-16, January 1975.</a>
+ * </p>
+ * 
+ * <p>
+ * This is a port from a python example by Michael Clerx, posted as an answer to
+ * a question about <a href=
+ * "http://stackoverflow.com/questions/1690953/transitive-reduction-algorithm-pseudocode">
+ * transitive reduction algorithm pseudocode</a> on
+ * <a href="http://stackoverflow.com">Stack Overflow</a>
+ * </p>
+ * 
+ * @author Christophe Thiebaud
+ * @since Aug 13, 2015
+ */
+
+public class TransitiveReduction<V, E> {
```

Comment:
Since there's no longer a constructor, maybe add an INSTANCE singleton same as in TransitiveClosure?